WordPress/WordPressTest/LikeUserHelperTests.swift

```diff
@@ -0,0 +1,78 @@
+        var remoteUserDictionary: [String: Any] = [
+            "ID": 15,
+            "login": "testlogin",
+            "name": "testname",
+            "site_ID": 20,
+            "avatar_URL": "wordpress.org/test2",
+            "bio": "testbio",
+            "date_liked": "2021-11-24T04:02:42+0000",
+        ]
+
+        if hasPreferredBlog {
+            remoteUserDictionary["preferred_blog"] = [
+                "id": 1,
+                "url": "wordpress.org/test1",
+                "name": "testblog",
+                "icon": [
+                    "img": "someimage.jpg",
+                ]
+            ]
+        }
+
+        return remoteUserDictionary
+    }
+
+    func testNewLikeUserWithPreferredBlog() {
+        let completionExpectation = expectation(description: "We expect the context to save successfully")
+        let contextManager = TestContextManager()
+        let context = contextManager.mainContext
+
+        let remoteUserDictionary = createTestRemoteUserDictionary(withPreferredBlog: true)
+        let remoteUser = RemoteLikeUser(dictionary: remoteUserDictionary, commentID: 25, siteID: 30)
+        let likeUser = LikeUserHelper.createOrUpdateFrom(remoteUser: remoteUser, context: context)
+        XCTAssertNotNil(likeUser)
+
+        XCTAssertNoThrow(contextManager.save(context, withCompletionBlock: {
+            completionExpectation.fulfill()
+        }))
+
+        waitForExpectations(timeout: 0.1)
+    }
+
+    func testUpdatingExistingUserToRemovePreferredBlog() {
+        let completionExpectation = expectation(description: "We expect the context to save successfully")
+        let contextManager = TestContextManager()
+        let context = contextManager.mainContext
+
+        // First we create the pre-existing user, so we can later modify it to remove the preferred blog
+
+        let remoteUserDictionary = createTestRemoteUserDictionary(withPreferredBlog: true)
+        let remoteUser = RemoteLikeUser(dictionary: remoteUserDictionary, commentID: 25, siteID: 30)
+        let existingLikeUser = LikeUserHelper.createOrUpdateFrom(remoteUser: remoteUser, context: context)
+        guard let existingPreferredBlog = existingLikeUser.preferredBlog else {
+            XCTFail()
+            return
+        }
+
+        XCTAssertNotNil(existingLikeUser)
+        XCTAssertNoThrow(contextManager.save(context))
+
+        // Then we remove the preferred blog from the remote user, so we can save it again and make sure
+        // the preferred blog deletion works fine.
+        remoteUser.preferredBlog = nil
+        let updatedLikeUser = LikeUserHelper.createOrUpdateFrom(remoteUser: remoteUser, context: context)
+        XCTAssertNotNil(updatedLikeUser)
+        XCTAssertTrue(existingPreferredBlog.isDeleted)
+
+        XCTAssertNoThrow(contextManager.save(context, withCompletionBlock: {
+            completionExpectation.fulfill()
+        }))
```

Comment:
For reference, I also tried the following in an attempt to catch the ObjC `NSException`:

```swift
XCTAssertNoThrow(
    try WPException.objcTry {
        contextManager.save(context) {
            completionExpectation.fulfill()
        }
        self.waitForExpectations(timeout: 50.1)
    }
)
```

But even when wrapping the call inside the `WPException` wrapper (which converts raised ObjC `NSException` into thrown `Swift.Error`) and ensuring exceptions raised from `waitForExpectation` would also be caught (since that's the actual line on which the test runner crashes), it didn't seem enough to be able to catch the `NSException`

So after all I cave and say let's call it a day, revert this, add a comment, and let the test crash on failure for now. And keep the fix about "making our unit tests failing instead of crashing on CoreData save failure" for another PR, ince if seems like it would be more complicated fix than initially thought anyway.

```suggestion
        // Note: If the save fails, the current code will `@throw` an ObjC-based `NSException` and the test
        // runner will thus **crash** (as opposed to having the test properly report a failure.
        // Sadly, `XCTAssertNoThrow` is not enough to catch this (as it only catches `Swift.Error` `throw`n by Swift
        // or `(NSError**)` inout-returned by ObjC, not interupt-based `[NSException raise]`), and wrapping all
        // this within an `WPException.objcTry` to convert the raised `NSException` into a thrown `Swift.Error` does
        // not seem enough either – likely because the exception is thrown from a different thread?).
        // TODO: Fix this (maybe by making the `TestContextManager` not raise an `NSException` in the first place, as
        // opposed to the regular `ContextManager` used by the app?) to make the unit test fail properly instead of crashing.
        contextManager.save(context, withCompletionBlock: {
            completionExpectation.fulfill()
        }))
```